src/component/parser/json/converter/DiagramConverter.ts

```diff
@@ -104,26 +103,44 @@ export default class DiagramConverter {
     return false;
   }
 
-  private deserializeShape(shape: BPMNShape, findShapeElement: (bpmnElement: string) => ShapeBpmnElement): Shape | undefined {
-    const bpmnElement = findShapeElement(shape.bpmnElement);
+  private deserializeShape(bpmnShape: BPMNShape, findShapeElement: (bpmnElement: string) => ShapeBpmnElement): Shape | undefined {
+    const bpmnElement = findShapeElement(bpmnShape.bpmnElement);
     if (bpmnElement) {
-      const bounds = DiagramConverter.deserializeBounds(shape);
+      const bounds = DiagramConverter.deserializeBounds(bpmnShape);
 
       if (
         (bpmnElement instanceof ShapeBpmnSubProcess ||
           (bpmnElement instanceof ShapeBpmnCallActivity && bpmnElement.callActivityKind === ShapeBpmnCallActivityKind.CALLING_PROCESS)) &&
-        !shape.isExpanded
+        !bpmnShape.isExpanded
       ) {
         bpmnElement.markers.push(ShapeBpmnMarkerKind.EXPAND);
       }
 
       let isHorizontal;
       if (ShapeUtil.isPoolOrLane(bpmnElement.kind)) {
-        isHorizontal = shape.isHorizontal ?? true;
+        isHorizontal = bpmnShape.isHorizontal ?? true;
       }
 
-      const label = this.deserializeLabel(shape.BPMNLabel, shape.id);
-      return new Shape(shape.id, bpmnElement, bounds, label, isHorizontal);
+      const bpmnLabel = bpmnShape.BPMNLabel;
+      const label = this.deserializeLabel(bpmnLabel, bpmnShape.id);
+      const shape = new Shape(bpmnShape.id, bpmnElement, bounds, label, isHorizontal);
+      DiagramConverter.setColorExtensionsOnShape(shape, bpmnShape);
+
+      return shape;
+    }
+  }
+
+  // 'BPMN in Color' extensions with fallback to bpmn.io colors
+  private static setColorExtensionsOnShape(shape: Shape, bpmnShape: BPMNShape): void {
```

Comment:
**suggestion:** 

I think we don't need to check if `background-color`, `fill`, `border-color` and `stroke` are in `bpmnShape`, because even if there are `undefined`, with this implementation or the suggested implementation, the attributes of `shape.extensions` will be `undefined`.


```suggestion
  private static setColorExtensionsOnShape(shape: Shape, bpmnShape: BPMNShape): void {
      shape.extensions.fillColor = <string>bpmnShape['background-color'] ?? <string>bpmnShape['fill'];
      shape.extensions.strokeColor = <string>bpmnShape['border-color'] ?? <string>bpmnShape['stroke'];
  }
```

Reply:
The usage of the color extensions is very rare, so the idea was to not have properties with `undefined` values but to have no property in this case.
The model is not stored internally, so I will give it a try to your proposal to simplify the code here and for Edge.

Reply:
@csouchet the proposal cannot be implemented out of the box

```ts
const backgroundColor = bpmnShape['background-color'];
```
generates a TS error `TS7053: Element implicitly has an 'any' type because expression of type '"background-color"' can't be used to index type 'BPMNShape'.   Property 'background-color' does not exist on type 'BPMNShape'.`

I can ignore the error but the resulting code is less lisible and ignoring error is misleading
```ts
    // eslint-disable-next-line @typescript-eslint/ban-ts-comment -- explain why!
    // @ts-ignore
    shape.extensions.fillColor = <string>bpmnShape['background-color'] ?? <string>bpmnShape['fill'];
    // eslint-disable-next-line @typescript-eslint/ban-ts-comment -- explain why!
    // @ts-ignore
    shape.extensions.strokeColor = <string>bpmnShape['border-color'] ?? <string>bpmnShape['stroke'];
```